odl/set/sets.py

```diff
@@ -210,16 +209,21 @@ def __init__(self, length):
             The fixed length of the strings in this set. Must be
             positive.
         """
-        le = int(length)
-        if le <= 0:
+        length_ = int(length)
+        if length_ <= 0:
             raise ValueError('`length` {} is not positive.'.format(length))
-        self._length = le
+        self._length = length_
 
     @property
     def length(self):
         """The length attribute."""
         return self._length
 
+    @property
+    def ndim(self):
+        """Fake dim 1, needed for this to work as domain."""
```

Comment:
Better than saying that it is fake, please add a work on why it should be 1 for this to work.